src/generators/dom/index.js

```diff
@@ -258,111 +281,58 @@ export default function dom ( parsed, source, options, names ) {
 		function ${name} ( options ) {
 			options = options || {};
+			${generator.usesRefs ? `\nthis.refs = {}` : ``}
+			this._state = ${initialState};${templateProperties.computed ? `\napplyComputations( this._state, this._state, {} );` : ``}
 
-			var component = this;${generator.usesRefs ? `\nthis.refs = {}` : ``}
-			var state = ${initialState};${templateProperties.computed ? `\napplyComputations( state, state, {} );` : ``}
-
-			var observers = {
-				immediate: Object.create( null ),
-				deferred: Object.create( null )
-			};
-
-			var callbacks = Object.create( null );
-
-			function dispatchObservers ( group, newState, oldState ) {
-				for ( var key in group ) {
-					if ( !( key in newState ) ) continue;
-
-					var newValue = newState[ key ];
-					var oldValue = oldState[ key ];
-
-					if ( newValue === oldValue && typeof newValue !== 'object' ) continue;
-
-					var callbacks = group[ key ];
-					if ( !callbacks ) continue;
-
-					for ( var i = 0; i < callbacks.length; i += 1 ) {
-						var callback = callbacks[i];
-						if ( callback.__calling ) continue;
-
-						callback.__calling = true;
-						callback.call( component, newValue, oldValue );
-						callback.__calling = false;
-					}
-				}
-			}
-
-			this.fire = function fire ( eventName, data ) {
-				var handlers = eventName in callbacks && callbacks[ eventName ].slice();
-				if ( !handlers ) return;
-
-				for ( var i = 0; i < handlers.length; i += 1 ) {
-					handlers[i].call( this, data );
-				}
-			};
-
-			this.get = function get ( key ) {
-				return key ? state[ key ] : state;
-			};
-
-			this.set = function set ( newState ) {
-				${builders.set}
+			this._observers = {
+				pre: Object.create( null ),
+				post: Object.create( null )
 			};
 
-			this._mount = function mount ( target, anchor ) {
-				mainFragment.mount( target, anchor );
-			}
-
-			this.observe = function ( key, callback, options ) {
-				var group = ( options && options.defer ) ? observers.deferred : observers.immediate;
+			this._handlers = Object.create( null );
 
-				( group[ key ] || ( group[ key ] = [] ) ).push( callback );
+			this.get = get;
+			this.fire = fire;
+			this.observe = observe;
+			this.on = on;
+			this.set = set;
+			this.teardown = teardown;
 
-				if ( !options || options.init !== false ) {
-					callback.__calling = true;
-					callback.call( component, state[ key ] );
-					callback.__calling = false;
-				}
-
-				return {
-					cancel: function () {
-						var index = group[ key ].indexOf( callback );
-						if ( ~index ) group[ key ].splice( index, 1 );
-					}
-				};
-			};
-
-			this.on = function on ( eventName, handler ) {
-				var handlers = callbacks[ eventName ] || ( callbacks[ eventName ] = [] );
-				handlers.push( handler );
-
-				return {
-					cancel: function () {
-						var index = handlers.indexOf( handler );
-						if ( ~index ) handlers.splice( index, 1 );
-					}
-				};
-			};
+			this.root = options.root;
+			this.yield = options.yield;
```

Comment:
Just noticed now. Shouldn’t these be underscored, because they are internal implementation details, similar to `_fragment` and friends.